libs/dal/src/repositories/subscriber/subscriber.schema.ts

```diff
@@ -178,3 +181,7 @@ subscriberSchema.plugin(mongooseDelete, { deletedAt: true, deletedBy: true, over
 export const Subscriber =
   (mongoose.models.Subscriber as mongoose.Model<SubscriberDBModel>) ||
   mongoose.model<SubscriberDBModel>('Subscriber', subscriberSchema);
+
+function index(fields: IndexDefinition<SubscriberEntity>, options?: IndexOptions) {
+  subscriberSchema.index(fields, options);
```

Comment:
@rifont How about such an approach? while developing I noticed we don't have our indexes typed, meaning we can do indexed typos.
Doing this as a POC, I tried to make it a bit more generic but TS gave me problems :/

Reply:
`mongoose` is so poor when it comes to the types... yes the problems might be because we force some fields...
I like this approach 🙌

Reply:
I like it too, let's revisit generic typings as part of a holistic approach to strongly typing Mongoose. Ship it 🚀